File: tools/predict.R
library(tidyverse)
library(plyr)
library(drc)
library(nCal)

df <- read.csv('data.csv')

m2 = drm.fit(df$np~df$sf, dat = df)
predict(m2, newdata = data.frame(df$sf))
plot(m2, log="xy")

predict(m2, newdata = data.frame(sf = 300))
predict(m2, newdata = data.frame(sf = 1000))
predict(m2, newdata = data.frame(sf = 3000))   #  8 991 224 
predict(m2, newdata = data.frame(sf = 10000))  # 20 251 346

File: tools/predict.py
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import PolynomialFeatures

# Importing the dataset
datas = pd.read_csv('data.csv')
datas

X = datas.iloc[:, 0:1].values
y = datas.iloc[:, 1].values

poly = PolynomialFeatures(degree = 1)
X_poly = poly.fit_transform(X)

poly.fit(X_poly, y)
lin2 = LinearRegression()
lin2.fit(X_poly, y)

lin = LinearRegression()
lin.fit(X, y)

plt.scatter(X, y, color = 'blue')
plt.plot(X, lin.predict(X), color = 'red')
plt.title('Linear Regression')
plt.xlabel('SF')
plt.ylabel('numPersons')

plt.show()


plt.scatter(X, y, color = 'blue')
plt.plot(X, lin2.predict(poly.fit_transform(X)), color = 'red')
plt.title('Polynomial Regression')
plt.xlabel('SF')
plt.ylabel('numPersons')
plt.show()

data2 = pd.read_csv('data2.csv')
newX = data2.iloc[:, 0:1].values
newX

lin.predict(newX)
lin2.predict(poly.fit_transform(newX))
